pkg/operator/pdbcontroller/pdb_controller.go

```diff
@@ -0,0 +1,152 @@
+	operatorClient       v1helpers.OperatorClient
+	kubeClient           kubernetes.Interface
+	infrastructureLister configv1listers.InfrastructureLister
+	clusterTopology      configv1.TopologyMode
+	targetNamespace      string
+	pdbManifest          []byte
+	factory              *factory.Factory
+	eventRecorder        events.Recorder
+}
+
+var _ factory.Controller = &PDBController{}
+
+func NewPDBController(
+	name string,
+	operatorClient v1helpers.OperatorClient,
+	kubeClient kubernetes.Interface,
+	kubeInformers v1helpers.KubeInformersForNamespaces,
+	eventRecorder events.Recorder,
+	infrastructureLister configv1listers.InfrastructureLister,
+	pdbManifest []byte,
+	targetNamespace string) factory.Controller {
+	pdbc := &PDBController{
+		name:                 name,
+		operatorClient:       operatorClient,
+		kubeClient:           kubeClient,
+		infrastructureLister: infrastructureLister,
+		eventRecorder:        eventRecorder,
+		pdbManifest:          pdbManifest,
+		targetNamespace:      targetNamespace,
+	}
+	return factory.New().ResyncEvery(1*time.Minute).WithInformers(
+		kubeInformers.InformersFor(targetNamespace).Policy().V1().PodDisruptionBudgets().Informer(),
+		kubeInformers.InformersFor("").Core().V1().Nodes().Informer(),
+		operatorClient.Informer(),
+	).WithSync(pdbc.Sync).ToController(pdbc.name, eventRecorder.WithComponentSuffix(strings.ToLower(name)+"-controller"))
+}
+
+func (pdbc *PDBController) getTopologyMode() (configv1.TopologyMode, error) {
+	// right now cluster topology cannot change, infrastructure cr is immutable,
+	// so we set it once in order not to run api call each time
+	if pdbc.clusterTopology != "" {
+		klog.V(4).Infof("HA mode is: %s", pdbc.clusterTopology)
+		return pdbc.clusterTopology, nil
+	}
+
+	var err error
+	pdbc.clusterTopology, err = getControlPlaneTopology(pdbc.infrastructureLister)
+	if err != nil {
+		klog.Errorf("Failed to get topology mode %w ", err)
+		return "", err
+	}
+
+	klog.Infof("HA mode is: %s", pdbc.clusterTopology)
+
+	return pdbc.clusterTopology, nil
+}
+
+func (pdbc *PDBController) Sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	haTopologyMode, err := pdbc.getTopologyMode()
+	if err != nil {
+		return err
+	}
+
+	if haTopologyMode != configv1.HighlyAvailableTopologyMode {
+		return nil
+	}
+
+	if err := pdbc.ensureGuardingPDB(ctx, syncCtx.Recorder()); err != nil {
+		return err
+	}
+
+	if err != nil {
+		_, _, updateErr := v1helpers.UpdateStatus(pdbc.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:    "GuardControllerDegraded",
+			Status:  operatorv1.ConditionTrue,
+			Reason:  "Error",
+			Message: err.Error(),
+		}))
+		if updateErr != nil {
+			syncCtx.Recorder().Warning("PDBControllerUpdatingStatus", updateErr.Error())
+		}
+		return err
+	}
+
+	_, _, updateErr := v1helpers.UpdateStatus(pdbc.operatorClient,
+		v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:   "PDBControllerDegraded",
```

Comment:
Is there any reason you're setting Degraded here? Typically you would only need to set Available.